Fix JSON encoding of backslashes

ChristophKaser opened the pull request:
At the moment, a BrowserFunction cannot return a string containing a backslash when using SWT.EDGE:

VM374:1 Uncaught SyntaxError: Bad escaped character in JSON at position 4
    at JSON.parse (<anonymous>)
    at window.callJava (<anonymous>:2:13)

This is caused by a missing escape in JSON.stringify.

iloveeclipse commented:
@ChristophKaser : thanks for contribution. 

Could you please 
- create an issue with steps to reproduce
- refer to this issue in the commit message (amend commit and force push)
- add a regression test or manual test case if possible

ChristophKaser commented:
@iloveeclipse Thank you for your reply. I created issue #605 and added a UnitTest to demnostrate the problem. Please note that the unit test only fails (without the patch) if the system property org.eclipse.swt.browser.DefaultType=edge is set, otherwise it simply uses the IE browser implementation, which is not affected by the bug.

After several failed tries to sign the ECA (which I am pretty sure I already did ages ago), I created a new eclipse account and changed the commit mail addresses to use this account, so this check now succeeds as well.

ChristophKaser commented:
Any updates? Is there anything I can do so this can be merged?
Btw, I am positive the test failures are unrelated to my (really tiny) change.

mbooth101 commented:
Hi @ChristophKaser yes the failures are old, and unrelated to your change.

FWIW your change looks okay to me, but I am not a Windows person so I have been wondering why we don't run the browser test suite against both IE and Edge.

deepika-u commented:
Hi @ChristophKaser 
I am able to reproduce the problem in this issue and now after applying the patch seeing the expected output. 
My vote goes with the fix.

I have tried this on below environment : 
Eclipse SDK
Version: 2023-06 (4.28)
Build id: I20230424-1800
OS: Windows 11, v.10.0, x86_64 / win32
Java vendor: Eclipse Adoptium
Java runtime version: 17.0.6+10
Java version: 17.0.6

akurtakov commented:
@mbooth101 Please merge if/when you feel OK with it. I don't have access to Windows thus can not test the Edge integration .

niraj-modi reviewed (approved):
Code change looks good, thanks @deepika-u for the verification.

deepika-u commented:
Verified this on below build and looks good to me.

Eclipse SDK
Version: 2023-06 (4.28)
Build id: I20230515-1800
OS: Windows 11, v.10.0, x86_64 / win32
Java vendor: Eclipse Adoptium
Java runtime version: 17.0.6+10
Java version: 17.0.6
